### DISCUSSION TITLE
Invite distribution: undelegated identities should take precedence over the delegated ones

### DISCUSSION THREAD

**Discussion by ltraveler** [idena-network/idena-docs#127] [IIPs]
Change the current invite distribution method giving the precedence to undelegated identities over the delegated ones.
The whole amount of invites per epoch would be divided between delegated and undelegated identities.
In case of delegation the identity would take just a half of possible invitation/invitations based on the current invite distribution rules.
To be considered as undelegated, Identity after undelegation will have to wait 2 epochs plus the remaining period of the epoch in which the delegation took place to gain precedence over the delegated identities in the invite distribution process.

**Comment by ltraveler** [comment]
Another option could be do not give any invites at all to delegated identities. It also make sense because delegated identities do not have any server costs. So the invitation rewards helps independent node to pay server expenses and would stimulate farmers to decentralize their pools.

**Comment by Zen-44** [comment]
This would incentivize pools to run a single node for every identity, becoming a "shadow pool". The protocol tries to encourage pools to use the on-chain delegation system to strip their voting power without affecting their economics. 
Note that the invite reward system is now stake dependant so there are a lot of funds that a pool would want to claim.

**Reply by ltraveler** [reply]
There is no way to prevent shadow pools. In any elections you could buy votes there is no way to prevent that. If pools want to attack Oracles they would definitely do it and that attack wouldn't have any relation to this proposal. I mean the IDENA economical model becoming more and more complicated, I think the shadow pool factor doesn't make sense anymore. It's just regular corruption that exists everywhere without any relation to IDENA ecosystem.

There are more factors on the side of that kind of discrimination:
1. Running multiple nodes is more expensive process.
2. Each of the independent node would bring more decentralization to the network.
3. It is more complicated process to organize managing multiple nodes in compare to manage delegated pool.
4. Self invitation process would become more complicated because you would need to connect to each independent node and change private keys.
5. If you have a family pool you can easily run 4-5 nodes on your VPS without any problems but when you have hundreds of nodes in the pool that would make the farm process more complicated.
6. That will stop people to delegate themselves into public pools and motivate them to run independent nodes.

**Reply by Zen-44** [reply]
Shadow pools are not excluded completely, but the protocol heavily encourages using the on-chain delegation system by running one node for how many identities there are in the pool.

Farm owners running multiple nodes instead of just one is not making the network more decentralised since all those nodes would fall under the control of a single person.

**Reply by ltraveler** [reply]
I think your logic is incorrect because there is no difference between regular independent node and the "shadow pool node" technically it is the same thing. That means it is decentralization and the term "shadow pool" doesn't make any sense because there is no difference between them. Which means the Oracle protection from being attacked by centralized voting by multiple independent nodes should be in the field of the future evolution of the validation process, rewards distribution and so on. As I mentioned before, since pool owners start to see any more or less good self-profit in that kind of Oracle attacks they would definitely start to do that.
My point of view that the future evolution of the validation mechanic and rewards distribution should make the farming process very complicated and less profitable this is the only way to prevent Oracle attacks.
The invite distribution discrimination would make farming process less profitable and/or more complicated which is definitely a good thing.

**Reply by ubiubi2018** [reply]
I think the whole idena of delegation mechanism is to destinct between voting rights and rewards, assuming that majority of farmers is just economical motivated and doesnt care for votes. It makes no sense to make delegation less profitable, it only makes sense to give delegates less voting rights. That did allready hapoen.

**Reply by ltraveler** [reply]
I wouldn't agree with that. The core team have repeatedly stated that decentralization is one of the main goals. The strategy of making farm process more complicated and more expensive imo the correct way to restrain farm activities. The IIP that I have proposed would definitely decrease the incomes of current farms especially now, when we are knowing that self-invitation strategy is quite popular among farms.

**Reply by Zen-44** [reply]
It would decresease the income of pools delegated publicly without affecting pools of individual nodes (many individual nodes that are ran separately by one person).
It does indeed make it harder for pools, but it also encourages not using the deleagtion system.

I believe the delegation system should only have voting related restrictions while keeping the same economic incentive.

**Reply by ltraveler** [reply]
As I mentioned before the process of encouraging not using the delegation system is a positive factor.
Also it is not correct to say `individual nodes (many individual nodes that are ran separately by one person` since you run multiple nodes and pay to other people to pass validation and make flips you have no difference to individual nodes. The only way to prevent your activity is to make it more complicated and less profitable. For now the further ecosystem evolution should flow into running individual nodes and small family pools (which would be not a problem at all because you can easily run 4-5 nodes in parallel in one server) and to produce more complications and less economical profit for big pools. And the core team share the same point of view which they have repeatedly referred in chronicles.

**Reply by ubiubi2018** [reply]
Please do some research why the delegation mechanism was introduced in the first place

**Reply by ltraveler** [reply]
> Please do some research why the delegation mechanism was introduced in the first place

Please do some research of the historical differences between reward distribution, between network sizes, between pool sizes, between discrimination factors and you will see that currently posting the question `why the delegation mechanism was introduced in the first place` doesn't make much sense anymore and why the delegation mechanism was introduced in the past.

**Reply by ubiubi2018** [reply]
Wrong, sry

**Reply by Zen-44** [reply]
The delegation system is a way of preserving voting rights such that pools cannot influence votings (pools delegated on chain). Encouraging pools to not use this system and instead run a node for every identity is not a positive factor for the network because they would gain voting powers for every identity.

Right now pools can trade off their voting rights for more economic rewards (this is the purpose of the delegation system). By only affecting the delegation system with these kind of restrictions, the delegation system becomes less effective.

As a note, the quadratic mining rewards was, for example, aimed at reducing how much pools can earn without investment, but it affects the entire network meaning the delegation system preserves its benefits for pools.

Any change aimed to discourage pools should affect pools running more than one node for all identities and pools that use the delegation system the same way.

**Reply by ltraveler** [reply]
> The delegation system is a way

The delegation system was the way... Since the reward distribution model underwent substantial transformation, now the delegation system is a negative factor which helps farms to continue self-invitation process and save server costs.
The voting power is not the factor anymore because pools has been dramatically shrunk more over the proposed IIP would stimulate network members to run independent nodes and avoid using public pools which makes network more decentralized.
As I mentioned before if pools wants to attack Oracles they would do it without any relation to delegation mechanic they just don't see any profit yet.
For now, since we have integrated all of that latest IIPs the future evolution system would be to make farm process more expensive and complicated.
And it is obvious that for now the farm self-invitation process is a negative factor in the invite distribution model. The increasing server costs for farms will make farming less profitable which is a good factor.

**Reply by ubiubi2018** [reply]
Inconvenient truth: if someone does run a shadow farm now because he thinks it will give him economic advantage, than this is unfruitful strategy and will stay unprofitable forever. Shadow farming would be only useful if you would really want to attack a voting at early stage of the network. But as long as it is just about financial rewards, it wont give economic advantage for the forseeable future and it is highly unlikely that devs will ever reward independed nodes cryptonomically for beeing undelegated. 

The system just rewards people who have higher stake or who do better flips and additional flips who have good reporting skills. But it would be outright stupid to start to reward shadow farming with more invites or something similar. Shadowfarming was the status quo in 2020. That is where we came from, when one of the likely several shadowfarmers in 2020 didnt decide not to bother anymore with trying to hide but to show of that je was building 10% of all blocks. 

I think the delegation mechanism is one of the unique selling points of idena, because it creates more transparency. Most other identity-projects seem to prefer to claim that human farming doesnt exist at all, which is delusional.

I see no logic in abandoning a unique selling point. And also i see no economic benefit for shadow farming people on the horizon in idena. 

As long as there is no run on iwasm and also no  idena sharding, marketcap will probably stay much to low that it is not really worth to keep the extra expenses of shadow farming. On the other side once there is hopefully more demand for idena, that could be a time where contril over oracle votes could become economically more valuable, but at that time many more farms will compete with each other and none of those farms will provide a benefit to the users such big that it can defend its monopol or even ability to attack idena. So why change it and give shadow farmers more invites? 

Or do you run a shadow farm yourself?

**Comment by ltraveler** [comment]
> if someone does run a shadow farm

We've never get to know about it. 

> will stay unprofitable forever

Estimated expenditure would be: about 500 - 700$ for the server (once) + 10-15$ for Electricity, Internet and Static Ip (monthly) + regular expenses as usually farms do to manage delegated farm (epochally).

> advantage for the forseeable future and it is highly unlikely that devs will ever reward independed nodes cryptonomically for beeing undelegated.

`We believe that this is contrary to our goals and interferes with the solution of the scalability trilemma which requires excessive decentralization.`

> when one of the likely several shadow farmers in 2020 didn't decide not to bother anymore with trying to hide but to show of that je was building 10% of all blocks.

Times change, now in case if the IIP that I am proposing would be integrated the network will get more decentralization. Also, you have to keep in mind that apart from farms there are some big public pools which is also not good. The IIP that I am promoting will keep small pools working as it was before but big public pools would be definitely struggling and that would stimulate their members to undelegate themselves and start independent mining process.

> I think the delegation mechanism is one of the unique selling points of idena, because it creates more transparency. Most other identity-projects seem to prefer to claim that human farming doesn't exist at all, which is delusional.

I have the opposite opinion, since last changes in the rewards distribution model, discrimination factors and so on we more and more coming into the point when farming becoming just a regular  private corruption and the way to prevent that corruption is to make farm incomes lower and lower through the further flip making/reporting culture improvement (template flips is one of the topical issue), stimulate decentralization by motivating people running independent nodes, promoting somehow `idena.io` on flip images and so on.

> So why change it and give shadow farmers more invites?

I think it is incorrect to use `shadow farmers` terminology, because there is no way to detect shadow farm, there is no difference between regular nodes and shadow farmers. The correct way to think would be that the network will get more independent nodes which is imo a good factor. As I mentioned before if farms want to make an Oracle attack they will make it without any relation to my IIP. But my IIP will bring more decentralization to the network without any substantial risk to it.

**Reply by Zen-44** [reply]
> there is no difference between regular nodes and shadow farmers

Shadow farmers have multiple nodes that look like individual identities, but are in fact controlled by the same person. That is what the delegation system is trying to not encourage.

A change like this would not actually decentralize the network. If a pool with 1000 nodes decides to stop using the delegation system and instead runs 1000 nodes that look like regular nodes, it may look like we are more decentralized, but is that really decentralization? That just gave them 1000 votes the pool they did not have before.

This will give a false sentiment of decentralization at first when the nodes number starts increasing (if pools stop using the delegation system), but in reality having one farmer running more than one node is not helpful for the network.

**Reply by ltraveler** [reply]
I think there is one logic fallacy here. If someone wants to get 1000 votes someone would get it. There is no relation to delegation system at all. Moreover, based on the cost calculation that I have shown in my previous message the financial investment into getting multiple votes privileges is not so big.

For farmers the profit is the only motivating factor and that factor do not have any relationship to the IIP that I am proposing, more over my IIP would stimulate farmers to invest more into Idena eco-system by renting virtual dedicated servers for example. Apart from that big public pools gonna be shrinked and their users that now prefer to delegate themselves would be motivated to run their own nodes which definitely brings more decintralizaton into the network.

> This will give a false sentiment of decentralization at first when the nodes number starts increasing

The only positive factor that dilegation gives imo is that when farm is delegated it is possible to do statistical review and do some research of how new changes affect pools. For example delegation mechanism helps to find out which pools produce toxic flips and how new changes in flip mechanics affect them.

**Reply by ubiubi18** [reply]
honestly no idea what you are up to

**Reply by ltraveler** [reply]
> honestly no idea what you are up to

Be free to quote unclear part of the message
